src/services/submissionService.ts

```diff
@@ -171,21 +172,42 @@ Promise<Result<gqlTypes.Submission>> => {
             }
         }
 
+        // => After it's validated we should send already an okay request to the user
+
         const submissionRef = dbInstance.collection('submissions').doc()
         const newSubmissionId = submissionRef.id
         const newSubmission = mapGqlEntityToDbEntity(submissionInput, newSubmissionId)
         
-        await submissionRef.set(newSubmission)
+        const batch = dbInstance.batch()
```

Comment:
we should [use a transaction](https://firebase.google.com/docs/firestore/manage-data/transactions#passing_information_out_of_transactions) instead of batch. 
Batch will do it with one request which is nice, but it doesn't handle rolling back an action if any of them fails.